File: src/main/java/redgear/core/tile/TileEntitySmart.java
package redgear.core.tile;

import net.minecraft.nbt.NBTTagCompound;

/**
 * TileEntity that saves the username of the person who placed it. Best if used
 * in conjunction with BlockSmart
 * 
 * @author Blackhole
 * 
 */
public class TileEntitySmart extends TileEntityGeneric {
	public String ownerName = "";

	/**
	 * Don't forget to override this function in all children if you want more
	 * vars!
	 */
	@Override
	public void writeToNBT(NBTTagCompound tag) {
		super.writeToNBT(tag);
		tag.setString("ownerName", ownerName);
	}

	/**
	 * Don't forget to override this function in all children if you want more
	 * vars!
	 */
	@Override
	public void readFromNBT(NBTTagCompound tag) {
		super.readFromNBT(tag);
		ownerName = tag.getString("ownerName");
	}
}

File: src/main/java/redgear/core/tile/TileEntitySmart.scala
package redgear.core.tile

import net.minecraft.nbt.NBTTagCompound

/**
 * @author Blackhole
 *         Created on 10/12/2014.
 */
class TileEntitySmart extends TileEntityGeneric with Owned{


  /**
   * Don't forget to override this function in all children if you want more
   * vars!
   */
  override def writeToNBT(tag: NBTTagCompound) {
    super.writeToNBT(tag)
  }

  /**
   * Don't forget to override this function in all children if you want more
   * vars!
   */
  override def readFromNBT(tag: NBTTagCompound) {
    super.readFromNBT(tag)
  }
}
